hoomd/mpcd/update.py

```diff
@@ -0,0 +1,112 @@
+# Copyright (c) 2009-2024 The Regents of the University of Michigan.
+# Part of HOOMD-blue, released under the BSD 3-Clause License.
+
+r"""Reverse nonequilibrium shear flow of an MPCD fluid.
+
+Reverse nonequilibirum shear flow of an MPCD fluid is established by imposing
+a known momentum flux on the system and measuring the resulting shear rate. This
+provides a way to calculate the shear viscosity of the solvent.
+
+.. invisible-code-block: python
+
+    simulation = hoomd.util.make_example_simulation(mpcd_types=["A"])
+    simulation.operations.integrator = hoomd.mpcd.Integrator(dt=0.1)
+
+"""
+
+from . import _mpcd
+from hoomd.operation import Updater
+from hoomd.data.parameterdicts import ParameterDict
+from hoomd.logging import log
+
+
+class ReverseNonequilibriumShearFlow(Updater):
+    r"""Reverse Perturbation method to establish nonequilibrium shear flow.
+
+    "Craig M. Tenney and Edward J. Maginn. Limitations and recommendations for
+    the calculation of shear viscosity using reverse nonequilibrium molecular
+    dynamics. J. Chem. Phys. 132, 014103, 2010."
+    `<https://doi.org/10.1063/1.3276454>`_
+
+    "Florian Mueller-Plathe. Reversing the perturbation in nonequilibrium
+     molecular dynamics: An easy way to calculate the shear viscosity of
+     fluids. Phys. Rev. E, 59:4894-4898, May 1999."
+     `<https://doi.org/10.1103/PhysRevE.59.4894>`_
+
+    The simulation box is divided into a number of slabs in y-direction. Two
+    exchange slabs separated by a distance of Ly/2 are chosen (typically the
+    bottom and middle slabs for convenience). Particles are sorted into the
+    slabs based on their positions and candidate particles, in the two exchange
+    slabs, with momentum close to a specified target are selected for a
+    pairwise momentum exchange (of their x-components). In the lower slab,
+    particles with positive momentum in flow direction are selected while those
+    with negative momentum are selected in the upper slab.
+
+    To simulate the orginal implementation by
+    `Mueller-Plathe <https://doi.org/10.1103/PhysRevE.59.4894>`_, only the
+    fastest and slowest particles in the bottom and upper slabs, respectively,
+    should be swapped. This can be acieved by setting the `num_swaps` argument
+    to 1 and the `target_momentum` argument to a sufficiently large value.
```

Comment:
You haven't discussed the Maginn paper. This would be a good spot, since you can explain that it is also possible to select a smaller target momentum.